### PR TITLE
Fix situation when invalid parametr name is used

### PR DESCRIPTION
**Pull request by JanCaha**
So far this code is considered ok by the plugin:
```
##Layer a=vector
```
It event produces the corresponding R code, which however is not valid. 
```
Layer a  <- st_read("path/to/folder", layer = "layer_name", quiet = TRUE, stringsAsFactors = FALSE)
```
As variable names in R cannot contain whitespace.

The PR solves the issue by syntax error if there is a variable name containing whitespace.

I am not sure if this is an ideal way to handle this issue but it seems to make sense to me.

### CODE REVIEW

**Comment by nyalldawson** [comment]
This approach looks fine to me. I suspect we should extend it to cover other invalid characters too -- e.g. `-`. Or just totally restrict it to alphanumeric + "_" characters?

Can you rebase this off master to remove the extra commits and add a unit test to test_algorithm.py?

**Comment by JanCaha** [comment]
Good point, I will check for all invalid variable names in R and prepare tests for it.

File: processing_r/processing/utils.py [hunk 1]
```diff
@@ -166,6 +166,34 @@ def create_descriptive_name(name):
...
+        # only letters a-z, A-Z, numbers, dot and underscore
+        x = re.search("[a-zA-Z0-9\\._]+", variable)
+
+        result = True
+
+        if variable == x.group():
+            # cannot start with number or underscore
+            x = re.search("^[0-9|_]", variable)
+            if x:
+                result = False
```
Comment (nyalldawson) [inline processing_r/processing/utils.py]: Minor thing. But we should return here and avoid the extra expense of an unnecessary regex test on the next line

Comment (JanCaha) [reply processing_r/processing/utils.py]: These two checks can be simplified into a single regex. Hopefully, I managed to do it correctly.  Waiting for travis to check out.

**Pull request merged**